### PR TITLE
Add clear button to SelectOneAutoAdvanceWidget [WIP]

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
I had only added clear button support to `SelectOneWidget` earlier. This PR will add clear button to the `SelectOneAutoAdvanceWidget` as well. 

Screenshot_2020-09-17-15-27-43-703_org commcare dalvik debug

### CODE REVIEW

File: app/src/org/commcare/views/widgets/SelectOneAutoAdvanceWidget.java [hunk 1]
```diff
@@ -122,17 +123,30 @@ public SelectOneAutoAdvanceWidget(Context context, FormEntryPrompt prompt) {
 
                 addView(thisParentLayout);
             }
+            addClearButton(context, s != null && !prompt.isReadOnly());
         }
     }
 
+    private void addClearButton(Context context, boolean show) {
```
Comment (shubham1g5) [inline app/src/org/commcare/views/widgets/SelectOneAutoAdvanceWidget.java:130]: this code is exactly same as the `addClearButton` in `SelectWidget` and should be reused.

**Review by shubham1g5 — CHANGES_REQUESTED** [review]
Shivam would be great to have screenshot of the change in the PR as well.

File: app/src/org/commcare/views/widgets/SelectOneAutoAdvanceWidget.java (hunk 1 above)
Comment (ShivamPokhriyal) [reply app/src/org/commcare/views/widgets/SelectOneAutoAdvanceWidget.java:130]: Yeah, would it be okay to have this method in `QuestionWidget`?

Comment (shubham1g5) [reply app/src/org/commcare/views/widgets/SelectOneAutoAdvanceWidget.java:130]: you can create a new abstract class like `SelectQuestionWidget : QuestionWidget` to put the logic specific to selects.

**Comment by ShivamPokhriyal** [comment]
I'll add the clear button support to `ListWidget` in `2.51` cuz it's kinda really complex right now.

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by shubham1g5** [comment]
> I'll add the clear button support to ListWidget in 2.51
Suport that idea, as it's not a good practice to get new things in this late in QA anyway. Can you accordingly adjust the milestone and target to master.

**Comment by ShivamPokhriyal** [comment]
Closing this in favour of #2367

**Pull request closed without merging**